innerprod/simple/lwe_test.go

```diff
@@ -74,9 +74,9 @@ func TestSimple_LWE(t *testing.T) {
 // testVectorData returns random vectors x, y, each containing
 // elements up to the respective bound.
 // It also returns the dot product of the vectors.
-func testVectorData(len int, xBound, yBound *big.Int) (data.Vector, data.Vector, *big.Int) {
-	x, _ := data.NewRandomVector(len, sample.NewUniform(xBound))
-	y, _ := data.NewRandomVector(len, sample.NewUniform(yBound))
+func testVectorData(len int, boundX, boundY *big.Int) (data.Vector, data.Vector, *big.Int) {
+	x, _ := data.NewRandomVector(len, sample.NewUniformRange(new(big.Int).Neg(boundX), boundX))
```

Comment:
Sorry for being picky, but for the sake of consistency, could you first create `samplerX` and `samplerY` outside the calls to `NewRandomVector`? This is how you did it in `innerprod/fullysec/lwe_test.go`.

Reply:
The assertion assert.Equals(t, xy, xyDecrypted, ...) was giving an error when both values were zero. I do not understand why exactly.

Reply:
Weird. Can you still reproduce this? I thought maybe `assert.Equal` failed when comparing +0 and -0, but this doesn't seem to be the case.

Reply:
I will reproduce it in the next commit.

Reply:
Is it possible that one of them has a bigger backing slice than the other? If I'm correct, `math/big` automatically resizes the backing slices of `big.Int`s when necessary - e.g. during computation. Then both contain 0, but one contains it within a larger than necessary slice. It's likely that I'm wrong, but I've had similar issues in the C version with GMP's `mpz_t` ints.

Reply:
I pushed a version where this error is occurring.